### PR TITLE
Add parent abstract class for parsers

### PR DESCRIPTION
**Pull request by pdr-z**
Add DataParser abstract class, a parent class to abstract away common attributes and methods for both parsers

### CODE REVIEW

File: data2rdf/csv_parser.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import io
-import uuid
 
 import magic
 import pandas as pd
 
+from data2rdf.parser import DataParser
 
-class CSVParser:
+# import uuid
```
Comment (yoavnash) [inline data2rdf/csv_parser.py:8]: Instead of commenting out the code, you can remove it. We could go back to it via visiting an older commit.

**Review by yoavnash — CHANGES_REQUESTED** [review]
Have the changes were tested against the unit tests?

File: data2rdf/csv_parser.py (hunk 1 above)
Comment (pdr-z) [reply data2rdf/csv_parser.py]: Yes, the unittests are running successfully

Comment (yoavnash) [reply data2rdf/csv_parser.py]: Thanks

**Review by yoavnash — APPROVED** [review]

**Pull request merged**